### PR TITLE
Satellite support for token exchange URL

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

**Review by skorati — APPROVED** [review]
looks good.
Please open a GHE on issues we discussed and we can track there.

File: pkg/k8s_utils/secret.go [hunk 1]
```diff
@@ -31,46 +31,56 @@ import (
...
+// GetSecret ...
+func GetSecret(kc KubernetesClient, secretname, dataname string) (string, error) {
+	kc.logger.Info("Fetching config map")
+
+	clientset := kc.GetClientSet()
+	namespace := kc.GetNameSpace()
+
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), secretname, v1.GetOptions{})
+	if err != nil {
+		kc.logger.Error("Error fetching cluster-info configmap", zap.Error(err))
```
Comment (arahamad) [inline pkg/k8s_utils/secret.go:61]: can we make error msg generic as this is not a cluster-info cm , its secret which is passed to this method

File: pkg/k8s_utils/secret.go [hunk 2]
```diff
@@ -31,46 +31,56 @@ import (
...
 		kc.logger.Error("No data found in the secret")
-		return "", "", utils.Error{Description: fmt.Sprintf(utils.ErrEmptyDataInSecret, secretname)}
+		return "", utils.Error{Description: fmt.Sprintf(utils.ErrEmptyDataInSecret, secretname)}
 	}
 
 	byteData, ok := secret.Data[dataname]
 	if !ok {
-		kc.logger.Error("Expected data not found in the secret")
-		return "", "", utils.Error{Description: fmt.Sprintf(utils.ErrExpectedDataNotFound, dataname, secretname)}
+		kc.logger.Error("cluster-config.json is not present")
```
Comment (arahamad) [inline pkg/k8s_utils/secret.go:72]: same here to make error msg generic.

**Review by arahamad — CHANGES_REQUESTED** [review]
can you fix these comments

**Review by arahamad — APPROVED** [review]
lgtm

**Pull request merged**